clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -282,6 +274,20 @@ private long appendEosRecord(long offset, long timestamp, ByteBuffer key, ByteBu
         return crc;
     }
 
+    private long appendOldRecord(long offset, long timestamp, ByteBuffer key, ByteBuffer value) throws IOException {
```

Comment:
As we discussed offline, let's leave the names as is. The central problem is that the new message format for EOS also changes the structure of the messages themselves. Before the message set and the message had the same format. Now they don't. 

It would be nice to capture these semantics in the name. It would also be nice to capture which implementation is current. 

However, I am at a loss to come up with names that satisfy both requirements. I would vote for good semantic names. Jason votes for names which indicate the current version. But even if we pick one, we can't come up with satisfactory names.

Let's take this to the AK community and see if they are more creative than we are!